eZ/Publish/Core/FieldType/User/Value.php

```diff
@@ -71,6 +71,11 @@ class Value extends BaseValue
      */
     public $maxLogin;
 
+    /**
+     * @var string
+     */
+    public $plainPassword;
```

Comment:
hmm, this is intended to take input but never have output?
So basically a writeOnly attribute then...

I know we don't have it anywhere today, and unsure if this would be good DX, but maybe we should consider a InputStruct for this field type to be used on creation / updates.. ?


Otherwise people will be asking why they don't have any value on this when loading data..

Reply:
yup, it is write only, if I am not mistaken similar solution is suggested somewhere in symfony docs  (or at least it was some time ago). Input struct for it seems like a good solution though.

As for `UserHandler::create/update`, I would say yes, we should get rid of those methods in the end but not sure where we should handle cache if we do.

Reply:
Introduction of InputStruct seems a far-reaching change to me, given the current  philosophy of 'accept plain php values as input for field values'.
Otoh what is the valid range for those php values, and what happens when an out-of-range value is used by the unwary developer is still, imho, one of the less-documented and less-tested areas of the kernel...

Reply:
> As for UserHandler::create/update, I would say yes, we should get rid of those methods in the end but not sure where we should handle cache if we do.

Well if we plan to get rid of load* methods it would be easy. But if not... what about invalidating the cache in some field type layer? like field  type storage _(direct, or cache decorator we align the value object to be same across handler and field type or whatever..)_

Reply:
Not sure about `load*` methods as those seems to be pretty useful, unless we add some generic method to fetch content by fieldType values, right?

But I truly prefers to do those as a followup to it, if you think this is possible in that case.

Reply:
What you do is up to engineering, just wanted to check if there was a plan to be aware of when doing review here.

If you say it will be handled in follow up, that is ok answer (but then don't forget ;))

Reply:
> If you say it will be handled in follow up, that is ok answer without going into details on how _(but then don't forget ;))_

Added a follow up for persistence cache: [EZP-30929](https://jira.ez.no/browse/EZP-30929). I didn't mention `load*` there, but the issue is still in spec, so we can discover what's necessary.